### ISSUE TITLE
Fix ranges::begin() test (regression)

### ISSUE THREAD

**Issue by mgehre** [mgehre/llvm-project#5]
The following test from test/Sema/warn-lifetime-godbolt.cpp does not cause diagnostics anymore:
```
void example_1_1_3_c() {
  vector<int> ints{0, 1, 2, 3, 4, 5};

  auto even = [](int i) { return !(i % 2); };

  auto view = ints | view::filter(even);

  cout << *ranges::begin(view); // ok

  ints.push_back(6);            // A: invalidates view TODOexpected-note {{modified here}}

  cout << *ranges::begin(view); // TODOexpected-warning {{passing a dangling pointer as argument}}
}
```
It still worked on https://github.com/mgehre/clang 5bd87fb8497635ed0d377ebcf0c0f2ccff9b57e7.
